pkg/operator/etcdcertsigner/etcdcertsignercontroller.go

```diff
@@ -331,6 +343,53 @@ func (c *EtcdCertSignerController) createSecretForNode(node *corev1.Node) error
 	return nil
 }
 
+func createPeerDNSSubjectAlternativeNamesFromMemberList(etcdDiscoveryDomain string, etcdMembers []*etcdserverpb.Member, nodeInternalIPs []string) []string {
+	var found bool
+	var errs []string
+	hosts := []string{"localhost"}
+	for _, m := range etcdMembers {
+		// 4.3 naming
+		if strings.HasPrefix(m.Name, "etcd-member") {
+			found = true
+		}
+		if len(m.PeerURLs) > 0 && len(m.ClientURLs) > 0 {
+			for _, p := range m.PeerURLs {
+				peer, err := url.Parse(p)
+				if err != nil {
+					errs = append(errs, err.Error())
+					continue
+				}
+				if addr := net.ParseIP(peer.Hostname()); addr != nil {
+					continue
+				}
+				for _, clientUrl := range m.ClientURLs {
```

Comment:
I don't understand this 3rd level nesting of for loop. Why do we have to loop through all clientURLs for every peer URL?

Reply:
well it might not be completely necessary I was covering the fact that it is a list. I guess it is fair to consider a manual adjustment to client-urls a fail and only attempt top match [0]? We could fall back to wildcard in the case no match for [0].

https://github.com/etcd-io/etcd/blob/master/Documentation/op-guide/configuration.md#--advertise-client-urls

Reply:
OK. In general, we typically have only one peer URL and one client URL per member. So it is not a big deal. Even otherwise, we are looping through 3 members: the loops are not big (not in thousands). Looks good to me.

Reply:
yeah simplified the logic and only check [0] for each. This is best-effort if the config is not default then we fallback.